### PR TITLE
Throw `syntaxError` with `loc` property if such error was encountered

### LINKED ISSUE #115: throw `syntaxError` with `loc` property if such error was encountered

### PR DESCRIPTION
**Pull request by markelog**
Fixes #115

### CODE REVIEW

**Comment by hzoo** [comment]
looks good?

File: test/lib/Parser.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        delete visitorKeys.ClassProperty;
+    });
+
+    it('should iterate by types only', () => {
+        try {
+            new Parser().parse(code);
+            expect(false).to.equal(true);
+
+        } catch (e) {
+            expect(e.message).to.equal('Cannot iterate using ClassProperty');
```
Comment (zxqfox) [inline test/lib/Parser.js]: Oh, really?

**Comment by zxqfox** [comment]
Don't get why ClassProperty still uniterable, but LGTM.

**Comment by markelog** [comment]
@zxqfox cause of this - https://github.com/cst/cst/pull/116/files#diff-18f06f03761fb72832a33d01d66e130dR14 Ж)

**Comment by zxqfox** [comment]
@markelog Omg, I need glasses :neckbeard:

**Pull request merged**